pinot-controller/src/main/java/org/apache/pinot/controller/helix/SegmentStatusChecker.java

```diff
@@ -161,6 +172,9 @@ private void updateSegmentMetrics(String tableNameWithType, Context context) {
     int nOffline = 0; // Keeps track of number segments with no online replicas
     int nSegments = 0; // Counts number of segments
     for (String partitionName : idealState.getPartitionSet()) {
+      if (!segmentsExcludeReplaced.contains(partitionName)) {
```

Comment:
Just curious why not use `segmentsExcludeReplaced` directly in the for loop?

Reply:
Oh correct, that's better. Updated.